ops/charm.py

```diff
@@ -1588,17 +1796,17 @@ def locations(self) -> List[str]:
         """An accessor for the list of locations for a mount."""
         return self._locations
 
-    def __getattr__(self, name: str):
-        # TODO(benhoyt): this should just be a property "location"
-        if name == "location":
-            if len(self._locations) == 1:
-                return self._locations[0]
-            else:
-                raise RuntimeError(
-                    "container has more than one mount point with the same backing storage. "
-                    "Request .locations to see a list"
-                )
-        else:
-            raise AttributeError(
-                f"{self.__class__.__name__} has no such attribute: {name}!"
-            )
+    @property
```

Comment:
This is a backwards-incompatible change in that you can no longer assign to `.location`, but it seems like that would surely always be a bug anyway, since it wouldn't actually change the location, just make the metadata incorrect.

Reply:
Excellent, thanks for this fix.